### PR TITLE
Adding validation to Region class

### PR DESCRIPTION
**Pull request by spfink**
At least adds some validation for #217 

## Description
Adds validation to Region.of(String value) to throw on null, empty or blank strings.

## Motivation and Context
#217

## Testing
Added additional test class.

## Types of changes
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist

- [X] I have read the **CONTRIBUTING** document
- [X] Local run of `mvn install` succeeds
- [X] My code follows the code style of this project
- [X] My change requires a change to the Javadoc documentation
- [X] I have updated the Javadoc documentation accordingly
- [X] I have read the **README** document
- [X] I have added tests to cover my changes
- [X] All new and existing tests passed
- [ ] A short description of the change has been added to the **CHANGELOG**

## License

- [X] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

File: core/src/main/java/software/amazon/awssdk/core/regions/Region.java [hunk 1]
```diff
@@ -120,10 +121,11 @@ private Region(String value) {
      * For example, the following conditions will always evaluated to true:
      * {@code Region.of("us-east-1") == Region.of("us-east-1")}, {@code Region.US_EAST_1 == Region.of("us-east-1")}.</p>
      *
-     * @param value The name of the region to load.
+     * @param value The name of the region to load. Can't be null, empty or blank
      * @return The region associated with the provided name.
      */
     public static Region of(String value) {
+        Validate.notBlank(value, "Region value can't be null, empty or blank.");
```
Comment (kiiadi) [inline core/src/main/java/software/amazon/awssdk/core/regions/Region.java]: Would this be better as `Validate.paramNotBlank(value, "region");`

File: core/src/test/java/software/amazon/awssdk/core/regions/RegionTest.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+    @Test(expected = IllegalArgumentException.class)
+    public void of_ThrowsIllegalArgumentException_WhenEmptyString() {
+        Region.of("");
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void of_ThrowsIllegalArgumentException_WhenBlankString() {
+        Region.of(" ");
+    }
```
Comment (kiiadi) [inline core/src/test/java/software/amazon/awssdk/core/regions/RegionTest.java]: Happy day test?

File: core/src/main/java/software/amazon/awssdk/core/regions/Region.java (hunk 1 above)
Comment (kiiadi) [reply core/src/main/java/software/amazon/awssdk/core/regions/Region.java]: Oh - it doesn't exist - maybe it should

Comment (spfink) [reply core/src/main/java/software/amazon/awssdk/core/regions/Region.java]: What would paramNotBlank do that notBlank doesn't?

**Pull request merged**